iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/SecretsGenerationCheck.java

```diff
@@ -77,5 +136,67 @@ private static void checkRunInstruction(CheckContext ctx, RunInstruction runInst
 
     DETECTORS.forEach(
       detector -> detector.search(resolvedArgument).forEach(command -> ctx.reportIssue(command, MESSAGE)));
+
+    CURL_DETECTORS.forEach(
+      detector -> detector.search(resolvedArgument).forEach(command -> {
+        ArgumentResolution lastArgument = getLastArgument(command);
+        if (lastArgument.value().contains(":")) {
+          ctx.reportIssue(command, MESSAGE);
+        }
+      }));
+
+    ArgumentResolutionSplitter.splitCommands(resolvedArgument).elements()
+      .forEach(arguments -> checkHtpasswd(arguments, ctx));
+  }
+
+  private static void checkHtpasswd(List<ArgumentResolution> resolvedArgument, CheckContext ctx) {
+    HtpasswdDetector detector = HtpasswdDetector.create(resolvedArgument);
+    if (detector.isDetected()) {
+      ctx.reportIssue(new CommandDetector.Command(resolvedArgument), MESSAGE);
+    }
+  }
+
+  private static ArgumentResolution getLastArgument(CommandDetector.Command command) {
+    List<ArgumentResolution> arguments = command.getResolvedArguments();
+    return arguments.get(arguments.size() - 1);
+  }
+
+  private static class HtpasswdDetector {
```

Comment:
This was almost my first implementation, but then I had a problem with cyclomatic complexity.
Other commands (like `git` `ls`) also support the pattern that `-b -n` can be replaced by `-bn`.
So next time it will be easier to extract similar logic.

Another reason for a separate static class is SRP/granularity. `SecretsGenerationCheck` executes all checks using "high-level" `CommandDetector` DLS. The logic in `HtpasswdDetector` is "low-level". It deserves for at least separate static class.

Reply:
Fair point. The thing that I don't like about this approach is the creation of a new class object for every command encountered. What about still encapsulating this in a private class, which only has a static boolean method instead of a "public" constructor?